### ISSUE TITLE
lists of valid tests to be ported over to jenkins ci

### ISSUE THREAD

**Issue by battlemidget** [juju-solutions/bundle-canonical-kubernetes#630]
We don't support bundletester any longer and we already run some of the same tests in our ci (test_cdk.py) environment. There are a couple of tests that still need to be ported over to CI.

If possible, could we get a list of bt tests that need to be ported over to CI in order to remove the use of running bundletester.

Thanks!

/cc @Cynerva @kwmonroe @hyperbolic2346 @ktsakalozos

**Comment by kwmonroe** [comment]
@battlemidget you're right, [validation](https://github.com/juju-solutions/kubernetes-jenkins/blob/master/integration-tests/validation.py) covers lots of bundletester tests implicitly.  BT explicitly tests for things like verifying services, tls, kubeconfig, labels, etc, but issues there would cause validation failures if that stuff didn't work.

Here's what I think we're missing:

- [20-charm-validation](https://github.com/juju-solutions/bundle-canonical-kubernetes/blob/master/tests/20-charm-validation.py)
  - master cluster info (`test_cluster_info`)
    - optional? are we doing anything that would cause `kubectl cluster-info` to fail?
  - worker config (`test_labels`)
  - scale tests (`test_etcd_scale_on_master`)
  - actions (`test_service_restart` and `test_namespace_actions`)

- [30-unit-shuffle](https://github.com/juju-solutions/bundle-canonical-kubernetes/blob/master/tests/30-unit-shuffle.py)
  - scale and delete (`test_scale_master_and_worker_termination`)

- [40-security-check](https://github.com/juju-solutions/bundle-canonical-kubernetes/blob/master/tests/40-security-check.py)
  - verify wrong dashboard password, change pw, and new pw (`test_basic`)

**Comment by Cynerva** [comment]
> * 20-charm-validation
>   * master cluster info (test_cluster_info)
>     * optional? are we doing anything that would cause kubectl cluster-info to fail?

We can skip this one - it's just a generic way to test that `kubectl` works, which is something we test implicitly and quite thoroughly.

> * 30-unit-shuffle
>   * scale and delete (test_scale_master_and_worker_termination)

Worth noting that this has overlap with [validate_worker_master_removal](https://github.com/juju-solutions/kubernetes-jenkins/blob/b7a976679256616ba42bf1c53db6901fc4b901a7/integration-tests/validation.py#L389), but it does test things that we don't cover anywhere else - for example, scaling down to 0 masters.

---

I would also add the following tests to the list:

* bundle-canonical-kubernetes
  * [40-security-check.py](https://github.com/juju-solutions/bundle-canonical-kubernetes/blob/master/tests/40-security-check.py)
    * Cannot access apiserver without authentication (`test_unauthenticated`)
* layer-etcd
  * [10-deploy.py](https://github.com/juju-solutions/layer-etcd/blob/master/tests/10-deploy.py)
    * `snapd_refresh` config (`test_config_snapd_refresh`)
    * etcd scale testing (`test_node_scale`, `test_node_scale_down_members`)
  * [20-actions.py](https://github.com/juju-solutions/layer-etcd/blob/master/tests/20-actions.py)
    * `health` action (`test_health_check`)
    * `snapshot` and `restore` actions (`test_snapshot_restore`)

**Comment by tvansteenburgh** [comment]
@battlemidget Did we get all these, can we close this?

**Comment by battlemidget** [comment]
I don't think this is fully resolved, I need to look into this more
